tests/spec/fs.unlink.spec.js

```diff
@@ -104,3 +104,33 @@ describe('fs.unlink', function() {
     });
   });
 });
+
+/**
+ * fsPromises tests
+ */
+
+describe('fsPromises.unlink', function() {
+  beforeEach(util.setup);
+  afterEach(util.cleanup);
+
+  it('should not unlink directories', () => {
+    var fs = util.fs().promises;
+
+    return fs.mkdir('/mydir')
+      .then(() => {
+        return fs.unlink('/mydir')
```

Comment:
A tip: when you use fat arrow `=>` functions like this, if the only thing your function does is to return a value (e.g., it's an expression), you can get rid of the `{ return...}` and do this instead:

```js
return fs.mkdir('/mydir')
  .then(() => fs.unlink('/mydir'))
```

What you have now is fine, just an FYI for later.  Nice work.